packages/blocks/src/api/raw-handling/ms-strip-comments.js

```diff
@@ -0,0 +1,41 @@
+export default function( node ) {
+	// Only filter on DOM elements
+	if ( ! node.getAttribute ) {
+		return;
+	}
+
+	// Filter based on 'style' attribute
+	const style = node.getAttribute( 'style' );
+
+	// Convert "<a style='mso-comment-reference...'>text</a>" => "text"
+	if ( style && style.indexOf( 'mso-comment-reference' ) !== -1 ) {
+		node.outerHTML = node.innerHTML;
```

Comment:
Could we unwrap the node instead of setting `outerHTML`? There's a utility function in `@wordpress/dom`.